firmware/ec/src/devices/at45db.c

```diff
@@ -0,0 +1,284 @@
+/**
+* Copyright (c) 2017-present, Facebook, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the BSD-style license found in the
+* LICENSE file in the root directory of this source tree. An additional grant
+* of patent rights can be found in the PATENTS file in the same directory.
+*
+* This file is used as Device layer for AT45DB641E. Mainly it contains Data read,
+* Data write, Page erase, Status check functions, these functions are called by
+* littlefs filesystyem in order to perform read/write operation for data using SPI
+* interface. Also while post execution device and manufacturing id's of AT45DB641E
+* will be verified by probe function.
+*/
+
+#include "inc/devices/at45db.h"
+#include "inc/common/spibus.h"
+#include "inc/common/global_header.h"
+#include "inc/global/OC_CONNECT1.h"
+
+#define AT45DB_MANFACTURE_ID            0x1F
+#define AT45DB_DEVICE_ID                0x0028
+#define AT45DB_READY                    0x80 /* AT45DB Ready Value */
+#define AT45DB_STATUS_OPCODE            0xD7
+#define AT45DB_PAGE_ERASE_OPCODE        0x81
+#define AT45DB_PAGE_RD_OPCODE           0xD2
+#define AT45DB_SRAM_BUFF2_WR_OPCODE     0x87
+#define AT45DB_PAGE_WR_OPCODE           0x86
+#define AT45DB_DEVID_RD_OPCODE          0x9F
+
+#define AT45DB_STATUS_OPCODE_WR_COUNT   1
+#define AT45DB_DEVID_OPCODE_WR_COUNT    1
+#define AT45DB_ERASE_OPCODE_WR_COUNT    4
+#define AT45DB_DATA_WR_OPCODE_WR_COUNT  4
+#define AT45DB_DATA_RD_OPCODE_WR_COUNT  8
+#define AT45DB_STATUS_RD_BYTES          1
+#define AT45DB_DEVID_RD_BYTES           2
+
```

Comment:
definitions in alphabetical order

Reply:
As per our current coding guidelines the "#include"s are to be in alphabetical order . If the same needs to be extended for "#define"s as well then we need to change this as part of a separate request as a lot of files might need to be changed .

Reply:
yes there is little effort in changing files in this PR but is it not better that all defines are arranged in a similar manner in the entire codebase and the coding guidelines also be updated to reflect the same ? Please share your opinion on the same

Reply:
My opinion is that the improvement can be made in these files at this time.  Saying that we shouldn't do it for these files because it should be done for all files is a way of say that it'll never be done.

Reply:
Changed as requested